File: app/src/main/java/com/example/marosu/secretchat/base/BaseContract.java
package com.example.marosu.secretchat.base;

import android.arch.lifecycle.Lifecycle;

/**
 * Created by Marius-Andrei Rosu on 8/7/2017.
 */
public interface BaseContract {

    interface View {
    }

    interface Presenter<V extends View> {
        void attachLifecycle(Lifecycle lifecycle);

        void detachLifecycle(Lifecycle lifecycle);

        void attachView(V view);

        void detachView();

        V getView();

        boolean isViewAttached();

        void onPresenterDestroy();
    }
}

File: app/src/main/java/com/example/marosu/secretchat/base/BaseContract.kt
package com.example.marosu.secretchat.base

import android.arch.lifecycle.Lifecycle

/**
 * Created by Marius-Andrei Rosu on 8/29/2017.
 */
interface BaseContract {
    interface View

    interface Presenter<V : View> {
        fun attachLifecycle(lifecycle: Lifecycle)

        fun detachLifecycle(lifecycle: Lifecycle)

        fun attachView(view: V)

        fun detachView()

        fun isViewAttached(): Boolean

        fun onPresenterDestroy()
    }
}
